raspiraw.c

```diff
@@ -695,6 +861,80 @@ int main(int argc, const char** argv) {
 		return -2;
 	}
 
+
+	if (cfg.regs)
+	{
+		int r,b;
+		char *p,*q;
+
+		p=strtok(cfg.regs, ";");
+		while (p)
+		{
+			vcos_assert(strlen(p)>6);
+			vcos_assert(p[4]==',');
+			vcos_assert(strlen(p)%2);
+			p[4]='\0'; q=p+5;
+			sscanf(p,"%4x",&r);
+			while(*q)
+			{
+				vcos_assert(isxdigit(q[0]));
+				vcos_assert(isxdigit(q[1]));
+
+				sscanf(q,"%2x",&b);	
+				vcos_log_error("%04x: %02x",r,b);    
+
+				set_reg(sensor_mode, r, b);
+
+				++r;
+				q+=2;
+			}
+			p=strtok(NULL,";");
+		}
+	}
+
+	if (cfg.hinc >= 0)
+	{
+		// TODO: handle modes different to ov5647 as well
+		set_reg(sensor_mode, 0x3814, cfg.hinc);
+	}
+
+	if (cfg.vinc >= 0)
+	{
+		// TODO: handle modes different to ov5647 as well
+		set_reg(sensor_mode, 0x3815, cfg.vinc);
+	}
+
+	if (cfg.fps > 0)
+	{
+		// TODO: handle modes different to ov5647 as well
+		int h = get_reg(sensor_mode, 0x380E);
+		int l = get_reg(sensor_mode, 0x380F);
+                int n = ((h<<8) + l) * sensor_mode->framerate / cfg.fps;
+		set_reg(sensor_mode, 0x380E, n>>8);
+		set_reg(sensor_mode, 0x380F, n&0xFF);
+	}
+
+	if (cfg.width > 0)
+	{
+		sensor_mode->width = cfg.width;
+	}
+
+	if (cfg.height > 0)
+	{
+		sensor_mode->height = cfg.height;
+	}
+
+	if (cfg.vts > 0)
+	{
+		sensor_mode->min_vts = cfg.vts;
+	}
+
+	if (cfg.line > 0)
+	{
+		sensor_mode->line_time_ns = cfg.line;
+	}
+
+
```

Comment:
The update_regs call at line 949 has the potential to undo anything up to this point.
Probably better to alter update_regs to take in the cfg struct (as a const) and then apply all these settings there as one central point.